snap/local/register-device.sh

```diff
@@ -1,12 +1,29 @@
 #!/bin/sh -e
 
-CONFIGURATION_FILE_PATH=$SNAP_COMMON/configuration/device.yaml
+CONFIGURATION_FILE_PATH=${SNAP_COMMON}/configuration/device.yaml
 
-if [ ! -f "$CONFIGURATION_FILE_PATH" ]; then
-    echo "Configuration file '$CONFIGURATION_FILE_PATH' does not exist."
+if [ ! -f "${CONFIGURATION_FILE_PATH}" ]; then
+    echo "Configuration file '${CONFIGURATION_FILE_PATH}' does not exist."
     exit 1
 fi
 
-$SNAP/bin/cos-registration-agent --shared-data-path $SNAP_COMMON/rob-cos-shared-data --grafana-dashboards $SNAP_COMMON/configuration/grafana_dashboards --foxglove-studio-dashboards $SNAP_COMMON/configuration/foxglove_layouts setup -c $CONFIGURATION_FILE_PATH
+# Set the registration command args based on configuration
+REGISTRATION_CMD_ARGS=""
+
+# Check if grafan_dashboards directory exists in configuration
+if [ -d "${SNAP_COMMON}/configuration/grafana_dashboards" ]; then
+    REGISTRATION_CMD_ARGS="${REGISTRATION_CMD_ARGS} --grafana-dashboards ${SNAP_COMMON}/configuration/grafana_dashboards"
+fi
+
+# Check if foxglove_layouts directory exists in configuration
+if [ -d "${SNAP_COMMON}/configuration/foxglove_layouts" ]; then
+    REGISTRATION_CMD_ARGS="${REGISTRATION_CMD_ARGS} --foxglove-studio-dashboards ${SNAP_COMMON}/configuration/foxglove_layouts"
+fi
+
+# Build the registration command with the args
+REGISTRATION_CMD="${SNAP}/bin/cos-registration-agent --shared-data-path ${SNAP_COMMON}/rob-cos-shared-data"
+REGISTRATION_CMD="${REGISTRATION_CMD} ${REGISTRATION_CMD_ARGS} setup -c ${CONFIGURATION_FILE_PATH}"
+
+eval "${REGISTRATION_CMD}"
```

Comment:
Why going through the eval, why not calling the cos-registration-agent directly with the list of argument?

Reply:
you are right, it made sense before I splitted the args from CMD. Changed in 09b7ddab0a4b712750251386304ee8d38248d7e3